DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -178,34 +177,34 @@ function Set-TargetResource
 
             if (-not (Test-RuleProperties -FirewallRule $firewallRule @PSBoundParameters))
             {
-                # Effectively renaming DisplayGroup to Group
-                if ($DisplayGroup) {
-                    $null = $PSBoundParameters.Add('Group', $DisplayGroup)
-                }
-
                 Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                     $($LocalizedData.UpdatingExistingFirewallMessage) -f $Name
                     ) -join '')
 
-                # If the DisplayGroup is being changed then the rule
-                # Has to be removed and recreated because that is the
-                # Only way the DisplayGroup can be changed unfortunately.
-                # A change to the set-netfirewallrule has been requested:
-                # https://connect.microsoft.com/PowerShell/feedbackdetail/view/1970765/add-ability-to-change-firewall-displaygroup-in-set-netfirewallrule-cmdlet
+                # If the Group is being changed the the rule needs to be recreated
                 if ($PSBoundParameters.ContainsKey('Group') `
                     -and ($Group -ne $FirewallRule.Group)) {
-                    # Although it is possible to automatically remove and recreate
-                    # this rule, it introduces more problems than it solves.
-                    # So for now this will throw an error with the suggested solution.
-                    $errorId = 'CantChangeDisplayGroupError'
-                    $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
-                    $errorMessage = $($LocalizedData.CantChangeDisplayGroupError) -f $Name
-                    $exception = New-Object -TypeName System.InvalidOperationException `
-                        -ArgumentList $errorMessage
-                    $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
-                        -ArgumentList $exception, $errorId, $errorCategory, $null
-
-                    $PSCmdlet.ThrowTerminatingError($errorRecord)
+
+                    Remove-NetFirewallRule -Name $Name
+
+                    # Merge the existing rule values into the PSBoundParameters
+                    # so that it can be splatted.
+
+                    # Loop through each possible property and if it is not passed as a parameter
+                    # then set the PSBoundParameter property to the exiting rule value.
+                    'DisplayName','Group','Enabled','Action','Profile','Direction', `
```

Comment:
Couple of suggestions here.

1) Piping is actually slower then foreach, so I'd suggest changing this to foreach ($parameter in $list) which wills peed it up.

2) Move this to it's own array as a seperate variable so we can modify this as needed. Though it would be aweosme to autodiscover in some way. In theory we could run $parameters = (Get-Command Get-NetFirewallRule).parameters but you'll get way more stuff then you're intending to get.